Flatpak for Bustle

wjt opened the pull request:
This is taken directly from https://cgit.freedesktop.org/bustle/tree/flatpak?h=bustle-0.6

TingPing commented on org.freedesktop.Bustle.json:
```diff
@@ -0,0 +1,252 @@
+                {
+                    "type": "git",
+                    "url": "https://anongit.freedesktop.org/git/bustle.git",
+                    "branch": "bustle-0.6"
```
Needs to include a specific commit or specific tarball.

alexlarsson commented on org.freedesktop.Bustle.json:
```diff
@@ -0,0 +1,252 @@
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.haskell.org/platform/download/8.0.2/haskell-platform-8.0.2-unknown-posix--full-x86_64.tar.gz",
```
This is x86-64 only, so you need to create a flathub.json with skip-arches for the other arches.
Alternatively, you could have only-arch on this module and add version for the other arches.
Here is an example flathub.json: https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/master/flathub.json

wjt replied:
Okay. The Haskell Platform bundle appears to only be available for i386 and x86_64 so I guess I will need `{ "skip-arches": ["arm", "aarch64"]}` in any case.

wjt commented:
Updated to build on i386, skip aarch64 and arm, and point explicitly to 0.6.1 by tag and commit hash.

Barthalion commented:
Test build started [here](https://flathub.org/builds/#/builders/1/builds/170).

Barthalion commented:
Looks like our buildbot is unhappy about C-style comments (although I think it wasn't before). Poked @ramcq on IRC about it.

wjt commented:
Well, on the one hand, it's totally legit to complain about them: they are not valid JSON, and @ebassi tells me that json-glib only allows them by accident.

On the other hand, almost every manifest I've seen uses them. Maybe Flathub insisting that JSON means JSON will change that.

ebassi commented:
> Maybe Flathub insisting that JSON means JSON will change that.

It's *highly* unlikely.

Maybe we should use another format for user-writable manifests, and then "compile" it to JSON.

wjt commented:
Here's a technique I've used in other places where I have to write JSON by hand but want to add a comment. Valid JSON is (modulo edge cases, please don't tell me about weird whitespace or surrogate pairs) valid YAML, and YAML supports `# ` comments. So I just renamed the manifest, replaced `/* */` with `#`, and wrote a Python program to convert it back.

ramcq commented on flathub.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "skip-arches": ["arm", "aarch64"]
```
We've now added only-arches to this file syntax. \o/

ramcq commented on update-manifest.py:
```diff
@@ -0,0 +1,17 @@
+
+
+def main():
+    '''The master copy is only YAML so we can have comments.'''
```
We should now accept
/* multi-line
 * c-style
 * comments */

wjt commented:
@ramcq fixed, if you want to test the multi-line comment fix

Barthalion commented:
The [build](https://flathub.org/builds/#/builders/1/builds/175) succeeded – I'll merge it later today.

Barthalion commented:
And with small delay, the repo has been created. Thanks!
